src/js/events/SystemLogActions.js

```diff
@@ -0,0 +1,118 @@
+import {
+  REQUEST_SYSTEM_LOG_SUCCESS,
+  REQUEST_SYSTEM_LOG_ERROR,
+  REQUEST_PREVIOUS_SYSTEM_LOG_SUCCESS,
+  REQUEST_PREVIOUS_SYSTEM_LOG_ERROR
+} from '../constants/ActionTypes';
+import AppDispatcher from './AppDispatcher';
+import CookieUtils from '../utils/CookieUtils';
+import SystemLogUtil from '../utils/SystemLogUtil';
+
+/**
+ * Implementation of server sent event handling for
+ * https://github.com/dcos/dcos-log
+ */
+
+// Store of current open connections
+let sources = {};
+const SystemLogActions = {
+  subscribe(nodeID, options = {}) {
+    let {
+      subscriptionID
+    } = options;
+
+    let url = SystemLogUtil.getUrl(nodeID, options);
+    subscriptionID = subscriptionID || global.btoa(url);
+    let source = new EventSource(url, {
+      withCredentials: Boolean(CookieUtils.getUserMetadata())
+    });
+
+    source.addEventListener('message', function ({data, origin, eventPhase} = {}) {
+      if (origin !== global.location.origin && eventPhase !== EventSource.OPEN) {
+        // Event is not from same origin, or is not open anymore
+        return false;
+      }
+
+      let parsedData = JSON.parse(data);
+      AppDispatcher.handleServerAction({
+        type: REQUEST_SYSTEM_LOG_SUCCESS,
+        data: parsedData,
+        subscriptionID
+      });
+    }, false);
+
+    source.addEventListener('error', (event = {}) => {
+      AppDispatcher.handleServerAction({
+        type: REQUEST_SYSTEM_LOG_ERROR,
+        data: event,
+        subscriptionID
+      });
+    }, false);
+
+    // Unsubscribe if any open connection exists with the same ID
+    this.unsubscribe(subscriptionID);
+    sources[subscriptionID] = {source};
+
+    return subscriptionID;
+  },
+
+  unsubscribe(subscriptionID) {
+    if (sources[subscriptionID]) {
+      sources[subscriptionID].source.close();
+      delete sources[subscriptionID];
+    }
+  },
+
+  fetchLogRange(nodeID, options = {}) {
```

Comment:
This method is in large parts very similar to the `subscribe` one. any chance we can separate the respective event source handling so that the two methods only handle the data?